devito/arguments.py

```diff
@@ -193,14 +189,15 @@ class DimensionArgProvider(ArgumentProvider):
 
     def __init__(self, *args, **kwargs):
         super(DimensionArgProvider, self).__init__(*args, **kwargs)
-        self._value = self._default_value
 
     def reset(self):
-        self._value = self._default_value
+        for i in self.rtargs:
+            i.reset()
 
     @property
     def value(self):
-        return self._value
+        child_values = tuple([i.value for i in self.rtargs])
+        return child_values if all(i is not None for i in child_values) else None
```

Comment:
In what case can they all be ``None`` ?

Reply:
Each runtime argument is independent and gets its value independent of the others. When it doesn't have a value, it is `None`, so they can all be `None` if none of them have a value. Is that your question? 
The code above, however, returns a tuple of values if all arguments have a known value, otherwise it returns `None`.

Reply:
I **think** so... unless it's too complicated, could you sketch (pseudocode is fine) a concrete example in which this happens ?

Reply:
Let's say we are talking of the `time` dimension, it has three `Argument`s - `time_size`, `time_start`, `time_end`. If `time_size.value=100`, `time_start.value=50` and `time_end.value=60`, this method would return `(100, 50, 60)`. If any of them doesn't have a value (and hence is `None`), this method would return `None`.

Reply:
You wrote:

> Each runtime argument is independent and gets its value independent of the others. When it doesn't have a value, it is None, so they can all be None if none of them have a value. Is that your question?
> The code above, however, returns a tuple of values if all arguments have a known value, otherwise it returns None.
> 

Do you think it'd be useful to add it as a comment in the code ?